### PR TITLE
DOC: Add examples to ``np.char``

### PR DESCRIPTION
**Pull request by luxedo**
Adds examples to `np.char` module. Part of https://github.com/numpy/numpy/issues/21351.

Examples added:
* `array` - defchararray module
* ~~`isalpha` - ufunc_docstrings module~~
* `isspace` - ufunc_docstrings module
* ~~`mod` - strings module~~
* ~~`rfind` - strings module~~
* `splitlines` - strings module  but in char namespace 
* ~~`startswith` - strings module - example similar to `endswith`~~

### CODE REVIEW

**Comment by ngoldbaum** [comment]
Our plan is to eventually get rid of `np.char` in favor of `np.strings`. Can you focus your efforts on improving examples in `np.strings` instead?

**Comment by luxedo** [comment]
This PR fills all the missing examples for `np.char` and `np.strings`:
* `isalpha` - ufunc_docstrings module
* `isspace` - ufunc_docstrings module
* `mod` - strings module
* `rfind` - strings module
* `startswith` - strings module - example similar to endswith

The ones below are added only to to `np.char`:
* `splitlines` - strings module but in char namespace
* `array` - defchararray module

The name `np.char` showed up because of the script that we are using to dig for missing examples in the documentation:
https://gist.github.com/luxedo/c7ad85b8848136671d126cd7baa07990

File: numpy/_core/strings.py [hunk 1]
```diff
@@ -263,6 +270,20 @@ def rfind(a, sub, start=0, end=None):
...
+    Examples
+    --------
+
+    While `string.find` returns the lowest index:
+
+    >>> a = np.array(["very repetitive very repetitive very"])
+    >>> np.strings.find(a, "very")
+    array([0])
+
+    `string.rfind` returns the highes index:
```
Comment (yuanx749) [inline numpy/_core/strings.py:282]:
```suggestion
    `string.rfind` returns the highest index:
```

**Comment by luxedo** [comment]
Rebased after other merges added examples. This PR now contains:

* `array` - defchararray module
* `isspace` - ufunc_docstrings module
* `splitlines` - strings module but in char namespace

File: numpy/_core/strings.py [hunk 2]
```diff
@@ -1373,6 +1373,14 @@ def _splitlines(a, keepends=None):
     --------
     str.splitlines
 
+    Examples
```
Comment (mdhaber) [inline numpy/_core/strings.py:1376]: Unless there's some magic that uses the docstring of the private function in the public function, not sure if this is needed here?

**Review by mdhaber — COMMENTED** [review]
I'll go ahead and resolve the merge conflict.

**Comment by mdhaber** [comment]
@ngoldbaum I'm not familiar with the plan for `np.char`, so I can't say whether these are needed. I'm also not familiar with these functions, but they don't hinder my understanding of the rest of the docstring. LMK if anything else is needed or if it's OK to merge.

**Comment by ngoldbaum** [comment]
I don't think these hurt. Let's merge.

**Pull request merged**